  new --enable option so we can disable the check for UTF16 string literals.
  See comments in our libs/icu/configure.ac.

diff -ur icu-4.6.orig/source/configure.in icu-4.6/source/configure.in
--- icu-4.6.orig/source/configure.in	2010-12-02 17:56:45.000000000 +0100
+++ icu-4.6/source/configure.in	2010-12-02 21:22:51.000000000 +0100
@@ -1013,7 +1013,27 @@
 fi
 AC_SUBST(U_SIZEOF_WCHAR_T)
 
+AC_MSG_CHECKING([whether to check for UTF-16 string literal support])
+enabled=yes
+U_ENABLE_UTF16_STRING_CHECK=1
+AC_ARG_ENABLE(utf16-string-check,
+    [  --enable-utf16-string-check enable testing for UTF-16 string literal support [default=yes]],
+    [ case "${enableval}" in
+	     yes|"") enabled=yes; U_ENABLE_UTF16_STRING_CHECK=1 ;;
+	     no) enabled=no; U_ENABLE_UTF16_STRING_CHECK=0 ;;
+	     *) ;;
+    esac],
+)
+AC_MSG_RESULT($enabled)
+AC_SUBST(U_ENABLE_UTF16_STRING_CHECK)
+
 AC_MSG_CHECKING([for UTF-16 string literal support])
+if test $U_ENABLE_UTF16_STRING_CHECK = 0; then
+U_CHECK_UTF16_STRING=0
+U_CHECK_GNUC_UTF16_STRING=0
+CHECK_UTF16_STRING_RESULT="check disabled"
+
+else # do the UTF16 literal checks
 U_CHECK_UTF16_STRING=1
 CHECK_UTF16_STRING_RESULT="unknown"
 
@@ -1098,6 +1118,7 @@
         fi
     fi
 fi
+fi
 AC_SUBST(U_CHECK_GNUC_UTF16_STRING)
 
 AC_MSG_RESULT($CHECK_UTF16_STRING_RESULT)
